dcos_launch/platforms/aws.py

```diff
@@ -278,32 +278,23 @@ def wait_for_status_change(self, state_1, state_2):
         UPDATE_ROLLBACK_FAILED, UPDATE_ROLLBACK_COMPLETE
         UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS
         """
-        log.info('Waiting for status to change from {} to {}'.format(state_1, state_2))
+        log.info('Waiting for stack creation or update to complete')
 
         @retrying.retry(wait_fixed=60 * 1000,
                         retry_on_result=lambda res: res is False,
                         retry_on_exception=lambda ex: False)
         def wait_loop():
             stack_status = self.get_status()
-            if stack_status == state_2:
+            if stack_status in ['CREATE_COMPLETE', 'UPDATE_COMPLETE']:
                 return True
-            if stack_status != state_1:
+            if stack_status not in ['CREATE_IN_PROGRESS', 'UPDATE_IN_PROGRESS', 'UPDATE_COMPLETE_CLEANUP_IN_PROGRESS']:
```

Comment:
It seems you have missed a critical use case of this function: https://github.com/dcos/dcos-launch/blob/master/dcos_launch/aws.py#L88

In this case here, `DELETE_IN_PROGRESS` and `DELETE_COMPLETE` will raise an error, but are entirely expected.